tmt/base.py

```diff
@@ -1247,6 +1248,8 @@ class Test(
 
     serial_number: int = field(default=0, internal=True)
 
+    _original_environment: Optional[tmt.utils.Environment] = None
+
```

Comment:
Slight naming confusion, it is unclear if this one contains the fmf file environment, the cli values etc.

As for the implementation, I would prefer for it to be a `field(init=False)` instead of having an `Optional`